File: 01_Language/01_Functions/python/hash_final.py
# coding: utf-8

import hashlib


def hash_final(context, raw_output=False):
    if raw_output:
        return context.digest()
    else:
        return context.hexdigest()


if __name__ == '__main__':
   h = hashlib.sha1()
   h.update(b"The quick brown fox jumped over the lazy dog.")
   print(hash_final(h))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/HashFinal.java
package com.onevgo.functions;

import com.google.common.io.BaseEncoding;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class HashFinal {
    public static String hashFinal(MessageDigest digest) {
        return BaseEncoding.base16().encode(digest.digest()).toLowerCase();
    }

    public static void main(String[] args) throws NoSuchAlgorithmException {
        MessageDigest sha1 = MessageDigest.getInstance("sha1");
        sha1.update("The quick brown fox jumped over the lazy dog.".getBytes());
        System.out.println(hashFinal(sha1));
    }
}
